oshi-core/src/main/java/oshi/software/os/windows/WindowsOperatingSystem.java

```diff
@@ -155,49 +158,22 @@ private static String parseVersion(WmiResult<OSVersionProperty> versionInfo, int
         // http://msdn.microsoft.com/en-us/library/windows/desktop/ms724833%28v=vs.85%29.aspx
         boolean ntWorkstation = WmiUtil.getUint32(versionInfo, OSVersionProperty.PRODUCTTYPE,
                 0) == WinNT.VER_NT_WORKSTATION;
-        switch (major) {
-        case 10:
-            if (minor == 0) {
-                if (ntWorkstation) {
-                    version = "10";
-                } else {
-                    // Build numbers greater than 17762 is Server 2019 for OS
-                    // Version 10.0
-                    version = (ParseUtil.parseLongOrDefault(buildNumber, 0L) > 17762) ? "Server 2019" : "Server 2016";
-                }
-            }
-            break;
-        case 6:
-            if (minor == 3) {
-                version = ntWorkstation ? "8.1" : "Server 2012 R2";
-            } else if (minor == 2) {
-                version = ntWorkstation ? "8" : "Server 2012";
-            } else if (minor == 1) {
-                version = ntWorkstation ? "7" : "Server 2008 R2";
-            } else if (minor == 0) {
-                version = ntWorkstation ? "Vista" : "Server 2008";
-            }
-            break;
-        case 5:
-            if (minor == 2) {
-                if ((suiteMask & 0x00008000) != 0) { // VER_SUITE_WH_SERVER
-                    version = "Home Server";
-                } else if (ntWorkstation) {
-                    version = "XP"; // 64 bits
-                } else {
-                    version = User32.INSTANCE.GetSystemMetrics(WinUser.SM_SERVERR2) != 0 ? "Server 2003"
-                            : "Server 2003 R2";
-                }
-            } else if (minor == 1) {
-                version = "XP"; // 32 bits
-            } else if (minor == 0) {
-                version = "2000";
-            }
-            break;
-        default:
-            break;
+
+        StringBuilder verLookup = new StringBuilder(major + "." + minor);
+
+        if ((suiteMask & 0x00008000) != 0) { // VER_SUITE_WH_SERVER
+            verLookup.append(".HS");
```

Comment:
Putting this first in an if/else means the other ones won't be checked. I am not sure if this will prevent you from testing `ntWorkstation` on 6.x or 10.x.  Probably best to specifically check for `5.2` on this,..

Reply:
I added a check before checking for suiteMask